src/reactThisBindingIssueRule.ts

```diff
@@ -86,11 +104,29 @@ class ReactThisBindingIssueRuleWalker extends ErrorTolerantWalker {
     protected visitMethodDeclaration(node: ts.MethodDeclaration): void {
         // reset variable scope when we encounter a method. Start tracking variables that are instantiated
         // in scope so we can make sure new function instances are not passed as JSX attributes
+        if (this.isMethodBoundWithDecorators(node)) {
+            this.boundListeners = this.boundListeners.concat('this.' + node.name.getText());
+        }
         this.scope = new Scope(null);
         super.visitMethodDeclaration(node);
         this.scope = null;
     }
 
+    private isMethodBoundWithDecorators(node: ts.MethodDeclaration): boolean {
```

Comment:
instead of using `this.allowedDecorators`, if we pass it to this function, then it'd become pure, will change.